pkg/sql/parser/scan.go

```diff
@@ -520,8 +520,23 @@ func (s *Scanner) scanIdent(lval *sqlSymType) {
 	start := s.pos - 1
 	for ; isIdentMiddle(s.peek()); s.pos++ {
 	}
-	lval.str = Name(s.in[start:s.pos]).Normalize()
-	if id, ok := keywords[strings.ToUpper(lval.str)]; ok {
+	// We know that the identifier we've seen so far is ASCII, so we don't need
+	// to unicode normalize. Instead, just upper and lowercase
+	bLower := make([]byte, s.pos - start)
+	bUpper := make([]byte, s.pos - start)
```

Comment:
Perhaps we could lazily allocate these in the case where `s.in[start:s.pos]` is either all lowercase or all uppercase.